src/coq_elpi_HOAS.ml

```diff
@@ -580,10 +580,16 @@ let type_of_global state r = S.update_return engine state (fun x ->
 
 let body_of_constant state c = S.update_return engine state (fun x ->
   match Global.body_of_constant_body Library.indirect_accessor (Environ.lookup_constant c x.env) with
-  | Some (bo, ctx) ->
+  | Some (bo, delayed, ctx) ->
      let inst, ctx = UnivGen.fresh_instance_from ctx None in
      let bo = Vars.subst_instance_constr inst bo in
      let evd = Evd.merge_context_set Evd.univ_rigid x.evd ctx in
+     let evd = match delayed with
+     | Opaqueproof.DelayedMonomorphic () -> evd
+     | Opaqueproof.DelayedPolymorphic (_, ctx) ->
+      let ctx = Util.on_snd (Univ.subst_univs_level_constraints (Univ.make_instance_subst inst)) ctx in
+      Evd.merge_context_set Evd.univ_rigid x.evd ctx
```

Comment:
why `x.evd` and not `evd`? The line above your patch uses `x.evd` to obtain `evd`...

Reply:
You're right, I'll fix this.